File: app/src/main/java/axisimski/bigramtagger/toLatin.java
package axisimski.bigramtagger;

/**
 * Created by Alex on 3/7/2018.
 */

public class toLatin {

    //Convert from Russian Cyrillic
    public String convertRU(String cyr){

        String lat;

        //Lower case letters
        lat=cyr.replaceAll("а","a");
        lat=lat.replaceAll("б","b");
        lat=lat.replaceAll("в","v");
        lat=lat.replaceAll("г","g");
        lat=lat.replaceAll("д","d");
        lat=lat.replaceAll("е","e");
        lat=lat.replaceAll("ё","e");
        lat=lat.replaceAll("ж","zh");
        lat=lat.replaceAll("з","z");
        lat=lat.replaceAll("и","i");
        lat=lat.replaceAll("й","i");
        lat=lat.replaceAll("к","k");
        lat=lat.replaceAll("л","l");
        lat=lat.replaceAll("м","m");
        lat=lat.replaceAll("н","n");
        lat=lat.replaceAll("о","o");
        lat=lat.replaceAll("п","p");
        lat=lat.replaceAll("р","r");
        lat=lat.replaceAll("с","s");
        lat=lat.replaceAll("т","t");
        lat=lat.replaceAll("у","u");
        lat=lat.replaceAll("ф","f");
        lat=lat.replaceAll("х","kh");
        lat=lat.replaceAll("ц","ts");
        lat=lat.replaceAll("ч","ch");
        lat=lat.replaceAll("ш","sh");
        lat=lat.replaceAll("щ","shch");
        lat=lat.replaceAll("ъ","ie");
        lat=lat.replaceAll("ы","y");
        lat=lat.replaceAll("ь","-");
        lat=lat.replaceAll("э","e");
        lat=lat.replaceAll("ю","iu");
        lat=lat.replaceAll("я","ia");

        //Capital letters
        lat=lat.replaceAll("А","A");
        lat=lat.replaceAll("Б","B");
        lat=lat.replaceAll("В","V");
        lat=lat.replaceAll("Г","G");
        lat=lat.replaceAll("Д","D");
        lat=lat.replaceAll("Е","E");
        lat=lat.replaceAll("Ё","E");
        lat=lat.replaceAll("Ж","ZH");
        lat=lat.replaceAll("З","Z");
        lat=lat.replaceAll("И","I");
        lat=lat.replaceAll("Л","I");
        lat=lat.replaceAll("К","K");
        lat=lat.replaceAll("Л","L");
        lat=lat.replaceAll("М","M");
        lat=lat.replaceAll("Н","N");
        lat=lat.replaceAll("О","O");
        lat=lat.replaceAll("П","P");
        lat=lat.replaceAll("Р","R");
        lat=lat.replaceAll("С","S");
        lat=lat.replaceAll("Т","T");
        lat=lat.replaceAll("У","U");
        lat=lat.replaceAll("Ф","F");
        lat=lat.replaceAll("Х","KH");
        lat=lat.replaceAll("Ц","TS");
        lat=lat.replaceAll("Ч","CH");
        lat=lat.replaceAll("Ш","SH");
        lat=lat.replaceAll("Щ","SHCH");
        lat=lat.replaceAll("Ъ","IE");
        lat=lat.replaceAll("Ы","Y");
        lat=lat.replaceAll("Ь","-");
        lat=lat.replaceAll("Э","E");
        lat=lat.replaceAll("Ю","IU");
        lat=lat.replaceAll("Я","IA");


        return lat; //Return latinized string


    }

    //Convert from Bulgarian Cyrillic
    public String convertBG(String cyr){

        String lat;

        lat=cyr.replaceAll("а","a");
        lat=lat.replaceAll("б","b");
        lat=lat.replaceAll("в","v");
        lat=lat.replaceAll("г","g");
        lat=lat.replaceAll("д","d");
        lat=lat.replaceAll("е","e");
        lat=lat.replaceAll("ж","j");
        lat=lat.replaceAll("з","z");
        lat=lat.replaceAll("и","i");
        lat=lat.replaceAll("й","i");
        lat=lat.replaceAll("к","k");
        lat=lat.replaceAll("л","l");
        lat=lat.replaceAll("м","m");
        lat=lat.replaceAll("н","n");
        lat=lat.replaceAll("о","o");
        lat=lat.replaceAll("п","p");
        lat=lat.replaceAll("р","r");
        lat=lat.replaceAll("с","s");
        lat=lat.replaceAll("т","t");
        lat=lat.replaceAll("у","u");
        lat=lat.replaceAll("ф","f");
        lat=lat.replaceAll("х","h");
        lat=lat.replaceAll("ц","c");
        lat=lat.replaceAll("ч","ch");
        lat=lat.replaceAll("ш","sh");
        lat=lat.replaceAll("щ","sht");
        lat=lat.replaceAll("ъ","u'");
        lat=lat.replaceAll("ь","'");
        lat=lat.replaceAll("ю","iu");
        lat=lat.replaceAll("я","ia");


        lat=lat.replaceAll("А","A");
        lat=lat.replaceAll("Б","B");
        lat=lat.replaceAll("В","V");
        lat=lat.replaceAll("Г","G");
        lat=lat.replaceAll("Д","D");
        lat=lat.replaceAll("Е","E");
        lat=lat.replaceAll("Ж","J");
        lat=lat.replaceAll("З","Z");
        lat=lat.replaceAll("И","I");
        lat=lat.replaceAll("Л","I");
        lat=lat.replaceAll("К","K");
        lat=lat.replaceAll("Л","L");
        lat=lat.replaceAll("М","M");
        lat=lat.replaceAll("Н","N");
        lat=lat.replaceAll("О","O");
        lat=lat.replaceAll("П","P");
        lat=lat.replaceAll("Р","R");
        lat=lat.replaceAll("С","S");
        lat=lat.replaceAll("Т","T");
        lat=lat.replaceAll("У","U");
        lat=lat.replaceAll("Ф","F");
        lat=lat.replaceAll("Х","H");
        lat=lat.replaceAll("Ц","C");
        lat=lat.replaceAll("Ч","CH");
        lat=lat.replaceAll("Ш","SH");
        lat=lat.replaceAll("Щ","SHT");
        lat=lat.replaceAll("Ъ","U'");
        lat=lat.replaceAll("Ь","'");
        lat=lat.replaceAll("Ю","IU");
        lat=lat.replaceAll("Я","IA");


        return lat;


    }

    //Convert from Mongolian Cyrillic
    public String convertMO(String cyr){

        String lat;

        lat=cyr.replaceAll("а","a");
        lat=lat.replaceAll("б","b");
        lat=lat.replaceAll("в","v");
        lat=lat.replaceAll("г","g");
        lat=lat.replaceAll("д","d");
        lat=lat.replaceAll("е","je");
        lat=lat.replaceAll("ё","jo");
        lat=lat.replaceAll("ж","zh");
        lat=lat.replaceAll("з","z");
        lat=lat.replaceAll("и","i");
        lat=lat.replaceAll("й","i");
        lat=lat.replaceAll("к","k");
        lat=lat.replaceAll("л","l");
        lat=lat.replaceAll("м","m");
        lat=lat.replaceAll("н","n");
        lat=lat.replaceAll("о","o");
        lat=lat.replaceAll("ө","o");
        lat=lat.replaceAll("п","p");
        lat=lat.replaceAll("р","r");
        lat=lat.replaceAll("с","s");
        lat=lat.replaceAll("т","t");
        lat=lat.replaceAll("у","u");
        lat=lat.replaceAll("ү","u");
        lat=lat.replaceAll("ф","f");
        lat=lat.replaceAll("х","h");
        lat=lat.replaceAll("ц","ts");
        lat=lat.replaceAll("ч","ch");
        lat=lat.replaceAll("ш","sh");
        lat=lat.replaceAll("щ","sht");
        lat=lat.replaceAll("ы","y");
        lat=lat.replaceAll("ь","'");
        lat=lat.replaceAll("ю","iu");
        lat=lat.replaceAll("я","ia");
        lat=lat.replaceAll("э","e");


        //Capital letters
        lat=lat.replaceAll("А","A");
        lat=lat.replaceAll("Б","B");
        lat=lat.replaceAll("В","V");
        lat=lat.replaceAll("Г","G");
        lat=lat.replaceAll("Д","D");
        lat=lat.replaceAll("Е","JE");
        lat=lat.replaceAll("Ё","JO");
        lat=lat.replaceAll("Ж","ZH");
        lat=lat.replaceAll("З","Z");
        lat=lat.replaceAll("И","I");
        lat=lat.replaceAll("Л","I");
        lat=lat.replaceAll("К","K");
        lat=lat.replaceAll("Л","L");
        lat=lat.replaceAll("М","M");
        lat=lat.replaceAll("Н","N");
        lat=lat.replaceAll("О","O");
        lat=lat.replaceAll("Ө","O");
        lat=lat.replaceAll("П","P");
        lat=lat.replaceAll("Р","R");
        lat=lat.replaceAll("С","S");
        lat=lat.replaceAll("Т","T");
        lat=lat.replaceAll("У","U");
        lat=lat.replaceAll("ү","U");
        lat=lat.replaceAll("Ф","F");
        lat=lat.replaceAll("Х","KH");
        lat=lat.replaceAll("Ц","TS");
        lat=lat.replaceAll("Ч","CH");
        lat=lat.replaceAll("Ш","SH");
        lat=lat.replaceAll("Щ","SHCH");
        lat=lat.replaceAll("Ъ","IE");
        lat=lat.replaceAll("Ы","Y");
        lat=lat.replaceAll("Ь","-");
        lat=lat.replaceAll("Э","E");
        lat=lat.replaceAll("Ю","IU");
        lat=lat.replaceAll("Я","IA");

        return lat;


    }





}

File: app/src/main/java/axisimski/bigramtagger/toLatin.kt
package axisimski.bigramtagger

/**
 * Created by Alex on 3/7/2018.
 */

class toLatin {

    //Convert from Russian Cyrillic
    fun convertRU(cyr: String): String {

        var lat: String

        //Lower case letters
        lat = cyr.replace("а".toRegex(), "a")
        lat = lat.replace("б".toRegex(), "b")
        lat = lat.replace("в".toRegex(), "v")
        lat = lat.replace("г".toRegex(), "g")
        lat = lat.replace("д".toRegex(), "d")
        lat = lat.replace("е".toRegex(), "e")
        lat = lat.replace("ё".toRegex(), "e")
        lat = lat.replace("ж".toRegex(), "zh")
        lat = lat.replace("з".toRegex(), "z")
        lat = lat.replace("и".toRegex(), "i")
        lat = lat.replace("й".toRegex(), "i")
        lat = lat.replace("к".toRegex(), "k")
        lat = lat.replace("л".toRegex(), "l")
        lat = lat.replace("м".toRegex(), "m")
        lat = lat.replace("н".toRegex(), "n")
        lat = lat.replace("о".toRegex(), "o")
        lat = lat.replace("п".toRegex(), "p")
        lat = lat.replace("р".toRegex(), "r")
        lat = lat.replace("с".toRegex(), "s")
        lat = lat.replace("т".toRegex(), "t")
        lat = lat.replace("у".toRegex(), "u")
        lat = lat.replace("ф".toRegex(), "f")
        lat = lat.replace("х".toRegex(), "kh")
        lat = lat.replace("ц".toRegex(), "ts")
        lat = lat.replace("ч".toRegex(), "ch")
        lat = lat.replace("ш".toRegex(), "sh")
        lat = lat.replace("щ".toRegex(), "shch")
        lat = lat.replace("ъ".toRegex(), "ie")
        lat = lat.replace("ы".toRegex(), "y")
        lat = lat.replace("ь".toRegex(), "-")
        lat = lat.replace("э".toRegex(), "e")
        lat = lat.replace("ю".toRegex(), "iu")
        lat = lat.replace("я".toRegex(), "ia")

        //Capital letters
        lat = lat.replace("А".toRegex(), "A")
        lat = lat.replace("Б".toRegex(), "B")
        lat = lat.replace("В".toRegex(), "V")
        lat = lat.replace("Г".toRegex(), "G")
        lat = lat.replace("Д".toRegex(), "D")
        lat = lat.replace("Е".toRegex(), "E")
        lat = lat.replace("Ё".toRegex(), "E")
        lat = lat.replace("Ж".toRegex(), "ZH")
        lat = lat.replace("З".toRegex(), "Z")
        lat = lat.replace("И".toRegex(), "I")
        lat = lat.replace("Л".toRegex(), "I")
        lat = lat.replace("К".toRegex(), "K")
        lat = lat.replace("Л".toRegex(), "L")
        lat = lat.replace("М".toRegex(), "M")
        lat = lat.replace("Н".toRegex(), "N")
        lat = lat.replace("О".toRegex(), "O")
        lat = lat.replace("П".toRegex(), "P")
        lat = lat.replace("Р".toRegex(), "R")
        lat = lat.replace("С".toRegex(), "S")
        lat = lat.replace("Т".toRegex(), "T")
        lat = lat.replace("У".toRegex(), "U")
        lat = lat.replace("Ф".toRegex(), "F")
        lat = lat.replace("Х".toRegex(), "KH")
        lat = lat.replace("Ц".toRegex(), "TS")
        lat = lat.replace("Ч".toRegex(), "CH")
        lat = lat.replace("Ш".toRegex(), "SH")
        lat = lat.replace("Щ".toRegex(), "SHCH")
        lat = lat.replace("Ъ".toRegex(), "IE")
        lat = lat.replace("Ы".toRegex(), "Y")
        lat = lat.replace("Ь".toRegex(), "-")
        lat = lat.replace("Э".toRegex(), "E")
        lat = lat.replace("Ю".toRegex(), "IU")
        lat = lat.replace("Я".toRegex(), "IA")


        return lat //Return latinized string


    }

    //Convert from Bulgarian Cyrillic
    fun convertBG(cyr: String): String {

        var lat: String

        lat = cyr.replace("а".toRegex(), "a")
        lat = lat.replace("б".toRegex(), "b")
        lat = lat.replace("в".toRegex(), "v")
        lat = lat.replace("г".toRegex(), "g")
        lat = lat.replace("д".toRegex(), "d")
        lat = lat.replace("е".toRegex(), "e")
        lat = lat.replace("ж".toRegex(), "j")
        lat = lat.replace("з".toRegex(), "z")
        lat = lat.replace("и".toRegex(), "i")
        lat = lat.replace("й".toRegex(), "i")
        lat = lat.replace("к".toRegex(), "k")
        lat = lat.replace("л".toRegex(), "l")
        lat = lat.replace("м".toRegex(), "m")
        lat = lat.replace("н".toRegex(), "n")
        lat = lat.replace("о".toRegex(), "o")
        lat = lat.replace("п".toRegex(), "p")
        lat = lat.replace("р".toRegex(), "r")
        lat = lat.replace("с".toRegex(), "s")
        lat = lat.replace("т".toRegex(), "t")
        lat = lat.replace("у".toRegex(), "u")
        lat = lat.replace("ф".toRegex(), "f")
        lat = lat.replace("х".toRegex(), "h")
        lat = lat.replace("ц".toRegex(), "c")
        lat = lat.replace("ч".toRegex(), "ch")
        lat = lat.replace("ш".toRegex(), "sh")
        lat = lat.replace("щ".toRegex(), "sht")
        lat = lat.replace("ъ".toRegex(), "u'")
        lat = lat.replace("ь".toRegex(), "'")
        lat = lat.replace("ю".toRegex(), "iu")
        lat = lat.replace("я".toRegex(), "ia")


        lat = lat.replace("А".toRegex(), "A")
        lat = lat.replace("Б".toRegex(), "B")
        lat = lat.replace("В".toRegex(), "V")
        lat = lat.replace("Г".toRegex(), "G")
        lat = lat.replace("Д".toRegex(), "D")
        lat = lat.replace("Е".toRegex(), "E")
        lat = lat.replace("Ж".toRegex(), "J")
        lat = lat.replace("З".toRegex(), "Z")
        lat = lat.replace("И".toRegex(), "I")
        lat = lat.replace("Л".toRegex(), "I")
        lat = lat.replace("К".toRegex(), "K")
        lat = lat.replace("Л".toRegex(), "L")
        lat = lat.replace("М".toRegex(), "M")
        lat = lat.replace("Н".toRegex(), "N")
        lat = lat.replace("О".toRegex(), "O")
        lat = lat.replace("П".toRegex(), "P")
        lat = lat.replace("Р".toRegex(), "R")
        lat = lat.replace("С".toRegex(), "S")
        lat = lat.replace("Т".toRegex(), "T")
        lat = lat.replace("У".toRegex(), "U")
        lat = lat.replace("Ф".toRegex(), "F")
        lat = lat.replace("Х".toRegex(), "H")
        lat = lat.replace("Ц".toRegex(), "C")
        lat = lat.replace("Ч".toRegex(), "CH")
        lat = lat.replace("Ш".toRegex(), "SH")
        lat = lat.replace("Щ".toRegex(), "SHT")
        lat = lat.replace("Ъ".toRegex(), "U'")
        lat = lat.replace("Ь".toRegex(), "'")
        lat = lat.replace("Ю".toRegex(), "IU")
        lat = lat.replace("Я".toRegex(), "IA")


        return lat


    }

    //Convert from Mongolian Cyrillic
    fun convertMO(cyr: String): String {

        var lat: String

        lat = cyr.replace("а".toRegex(), "a")
        lat = lat.replace("б".toRegex(), "b")
        lat = lat.replace("в".toRegex(), "v")
        lat = lat.replace("г".toRegex(), "g")
        lat = lat.replace("д".toRegex(), "d")
        lat = lat.replace("е".toRegex(), "je")
        lat = lat.replace("ё".toRegex(), "jo")
        lat = lat.replace("ж".toRegex(), "zh")
        lat = lat.replace("з".toRegex(), "z")
        lat = lat.replace("и".toRegex(), "i")
        lat = lat.replace("й".toRegex(), "i")
        lat = lat.replace("к".toRegex(), "k")
        lat = lat.replace("л".toRegex(), "l")
        lat = lat.replace("м".toRegex(), "m")
        lat = lat.replace("н".toRegex(), "n")
        lat = lat.replace("о".toRegex(), "o")
        lat = lat.replace("ө".toRegex(), "o")
        lat = lat.replace("п".toRegex(), "p")
        lat = lat.replace("р".toRegex(), "r")
        lat = lat.replace("с".toRegex(), "s")
        lat = lat.replace("т".toRegex(), "t")
        lat = lat.replace("у".toRegex(), "u")
        lat = lat.replace("ү".toRegex(), "u")
        lat = lat.replace("ф".toRegex(), "f")
        lat = lat.replace("х".toRegex(), "h")
        lat = lat.replace("ц".toRegex(), "ts")
        lat = lat.replace("ч".toRegex(), "ch")
        lat = lat.replace("ш".toRegex(), "sh")
        lat = lat.replace("щ".toRegex(), "sht")
        lat = lat.replace("ы".toRegex(), "y")
        lat = lat.replace("ь".toRegex(), "'")
        lat = lat.replace("ю".toRegex(), "iu")
        lat = lat.replace("я".toRegex(), "ia")
        lat = lat.replace("э".toRegex(), "e")


        //Capital letters
        lat = lat.replace("А".toRegex(), "A")
        lat = lat.replace("Б".toRegex(), "B")
        lat = lat.replace("В".toRegex(), "V")
        lat = lat.replace("Г".toRegex(), "G")
        lat = lat.replace("Д".toRegex(), "D")
        lat = lat.replace("Е".toRegex(), "JE")
        lat = lat.replace("Ё".toRegex(), "JO")
        lat = lat.replace("Ж".toRegex(), "ZH")
        lat = lat.replace("З".toRegex(), "Z")
        lat = lat.replace("И".toRegex(), "I")
        lat = lat.replace("Л".toRegex(), "I")
        lat = lat.replace("К".toRegex(), "K")
        lat = lat.replace("Л".toRegex(), "L")
        lat = lat.replace("М".toRegex(), "M")
        lat = lat.replace("Н".toRegex(), "N")
        lat = lat.replace("О".toRegex(), "O")
        lat = lat.replace("Ө".toRegex(), "O")
        lat = lat.replace("П".toRegex(), "P")
        lat = lat.replace("Р".toRegex(), "R")
        lat = lat.replace("С".toRegex(), "S")
        lat = lat.replace("Т".toRegex(), "T")
        lat = lat.replace("У".toRegex(), "U")
        lat = lat.replace("ү".toRegex(), "U")
        lat = lat.replace("Ф".toRegex(), "F")
        lat = lat.replace("Х".toRegex(), "KH")
        lat = lat.replace("Ц".toRegex(), "TS")
        lat = lat.replace("Ч".toRegex(), "CH")
        lat = lat.replace("Ш".toRegex(), "SH")
        lat = lat.replace("Щ".toRegex(), "SHCH")
        lat = lat.replace("Ъ".toRegex(), "IE")
        lat = lat.replace("Ы".toRegex(), "Y")
        lat = lat.replace("Ь".toRegex(), "-")
        lat = lat.replace("Э".toRegex(), "E")
        lat = lat.replace("Ю".toRegex(), "IU")
        lat = lat.replace("Я".toRegex(), "IA")


        return lat

    }


}
